pages/article/ArticleHandler.inc.php

```diff
@@ -49,20 +78,82 @@ function authorize($request, &$args, $roleAssignments) {
+				2) /article/view/[articleId]/[galleyId] ->  /article/view/galley/[articleId]/[galleyId]
+			*/
+			switch(count($args)) {
+				case 1: // article
+					$request->redirect(null, 'article', 'view', array('article', $this->articleId));
+					break;
+				case 2: // galley
+					$this->galleyId = isset($args[1]) ? $args[1] : 0;
+					$request->redirect(null, 'article', 'view', array('galley', $this->articleId, $this->galleyId));
+					break;
+				default:
+					assert(false);
+			}
+		} else {
+			$this->type = $args[0];
+			$this->articleId = isset($args[1]) ? $args[1] : 0;
+			
+			$submissionDao = Application::getSubmissionDAO();
+			$this->latestSubmissionRevision = $submissionDao->getLatestRevisionId($this->articleId, $this->journal->getId());
+			
+			switch($this->type) {
+				case 'article':
+					$this->submissionRevision = isset($args[2]) ? $args[2] : 0;
+					break;
+				case 'galley':
+					$this->galleyId = isset($args[2]) ? $args[2] : 0;
+					$this->submissionRevision = isset($args[3]) ? $args[3] : $this->latestSubmissionRevision;
+					break;
+				case 'file':
+					$this->galleyId = isset($args[2]) ? $args[2] : 0;
+					$this->submissionRevision = isset($args[3]) ? $args[3] : $this->latestSubmissionRevision;
+					$this->fileId = isset($args[4]) ? $args[4] : 0;
+					$this->fileRevision = isset($args[5]) ? $args[5] : 0;
+					break;
+				default:
+					assert(false);
+			}
+			
+			if ($this->submissionRevision &&($this->submissionRevision < $this->latestSubmissionRevision)) {
+				$this->isPreviousRevision = true;
+			} else {
+				$this->previousRevisions = $submissionDao->getSubmissionRevisions($this->articleId, $this->journal->getId(), false, true);
+			}
+		}
+	}
 
 	/**
 	 * @see PKPHandler::initialize()
 	 */
 	function initialize($request, $args) {
-		$articleId = isset($args[0]) ? $args[0] : 0;
-		$galleyId = isset($args[1]) ? $args[1] : 0;
-
-		$this->journal = $request->getContext();
+		
+		$this->journal = $request->getContext(); 
+		$this->_initializeArguments($request, $args);
+		
 		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
 		if ($this->journal->getSetting('enablePublicArticleId')) {
-			$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId((int) $this->journal->getId(), $articleId, true);
+			$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId((int) $this->journal->getId(), $this->articleId, true);
 		} else {
-			$publishedArticle = $publishedArticleDao->getPublishedArticleByArticleId((int) $articleId, (int) $this->journal->getId(), true);
+			$publishedArticle = $publishedArticleDao->getPublishedArticleByArticleId((int) $this->articleId, (int) $this->journal->getId(), true);
+		}
+		
+		if ($this->submissionRevision) {
+			$publishedArticle = $publishedArticleDao->getPublishedArticleByArticleId((int) $publishedArticle->getId(), (int) $this->journal->getId(), false, $this->submissionRevision);
```

Comment:
If I enter a version that doesn't exist (...article/view/article/1/10) I get a strange web page...